crates/holochain_types/src/app/app_manifest/app_manifest_v1.rs

```diff
@@ -194,6 +194,44 @@ impl AppManifestV1 {
         }
     }
 
+    /// Selectively overrides the modifiers for the given roles. Only fields with value `Some(T)` will
+    /// override the corresponding value in the manifest. If `None` is provided for a modifier field
+    /// the corresponding value in the manifest will remain untouched.
+    pub fn override_modifiers(
+        &mut self,
+        modifiers: HashMap<RoleName, DnaModifiersOpt<YamlProperties>>,
+    ) -> AppManifestResult<()> {
+        let existing_role_names = self
+            .roles
+            .iter()
+            .map(|manifest| &manifest.name)
+            .collect::<Vec<&String>>();
+        for role_name in modifiers.keys() {
+            if !existing_role_names.contains(&role_name) {
+                return Err(AppManifestError::InvalidRoleName(format!(
+                    "Tried to set modifiers for a role name that does not exist in the dna manifest: {role_name}"
+                )));
+            }
+        }
+        for role in self.roles.iter_mut() {
+            if let Some(modifier_opts) = modifiers.get(&role.name) {
```

Comment:
Iterate the modifiers because we know all of them are in `self.roles`, so this check wouldn't be needed?

Reply:
I cannot decode what you mean here (probably due to unfamiliarity with Rust), can you elaborate?

Reply:
The modifiers is a subset of the roles and I was thinking that the roles on the manifest would have random access but they don't, it's a vector. If they were both random access, then iterating the smaller set would mean you could drop the `if` condition but that doesn't work so well here. Never mind!